### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by AndrewHudzenko**

### CODE REVIEW

File: src/main/java/mate/service/DriverServiceImpl.java [hunk 1]
```diff
@@ -11,6 +12,9 @@ public class DriverServiceImpl implements DriverService {
     @Inject
     private DriverDao driverDao;
 
+    public DriverServiceImpl(DriverDaoImpl driverDao) {
+    }
+
```
Comment (justpopova) [inline src/main/java/mate/service/DriverServiceImpl.java:17]: i don't that need this? don't know where u use it

File: src/main/java/mate/controller/CreateCarController.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public class CreateCarController extends HttpServlet {
+    private static final Injector injector
+            = Injector.getInstance("mate");
+    private static final CarService carService
+            = (CarService) injector.getInstance(CarService.class);
+    private static final ManufacturerService manufacturerService
+            = (ManufacturerService) injector.getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
```
Comment (justpopova) [inline src/main/java/mate/controller/CreateCarController.java:26]:
```suggestion
    public void doGet(HttpServletRequest req, HttpServletResponse resp)
```
in controllers u have this methods or public, or protected. choose one option)(check other controllers and their  access modifiers)

File: src/main/java/mate/controller/CreateDriverController.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Driver;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/createDriver")
+public class CreateDriverController extends HttpServlet {
```
Comment (justpopova) [inline src/main/java/mate/controller/CreateDriverController.java:14]:
```suggestion
public class AddDriverController extends HttpServlet {
```
maybe change this controller to begin with Add(as in checklist, or if u think it's ok, can leave like that)
- AddManufacturerController
-AddDriverController

File: src/main/java/mate/controller/DisplayAllCarsController.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.CarService;
+
+@WebServlet(urlPatterns = "/displayAllCars")
+public class DisplayAllCarsController extends HttpServlet {
```
Comment (justpopova) [inline src/main/java/mate/controller/DisplayAllCarsController.java:13]:
```suggestion
public class GetAllCarsController extends HttpServlet {
```
this also maybe change to GetAllCarsController, as in checklist
-GetAllDriversController
-GetAllManufacturersController

**Review by justpopova — CHANGES_REQUESTED** [review]
Good Job!

**Review by justpopova — APPROVED** [review]
🔥

File: src/main/java/mate/controller/AddDriverToCarController.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        req.getRequestDispatcher("WEB-INF/views/car/addDriverToCar.jsp");
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Driver driver = driverService.get(Long.valueOf(req.getParameter("driver_id")));
+        Car car = carService.get(Long.valueOf(req.getParameter("car_id")));
+        carService.addDriverToCar(driver, car);
+        resp.sendRedirect(req.getContextPath() + "/carsWithDrivers");
```
Comment (Noct2000) [inline src/main/java/mate/controller/AddDriverToCarController.java:36]: where do you have a controller with `/carsWithDrivers` URL?

File: src/main/java/mate/controller/AddCarController.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Manufacturer;
+import mate.service.CarService;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/addCar")
```
Comment (Noct2000) [inline src/main/java/mate/controller/AddCarController.java:16]: check all URLs in the `Good URLs naming in your project` in the common mistakes

File: src/main/java/mate/controller/IndexController.java [hunk 7]
```diff
@@ -2,14 +2,17 @@
...
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+@WebServlet(urlPatterns = "/")
 public class IndexController extends HttpServlet {
     @Override
-    public void doGet(HttpServletRequest req, HttpServletResponse resp)
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
+        req.setAttribute("message", "WELCOME TO OUT TAXI SERVICE");
```
Comment (Noct2000) [inline src/main/java/mate/controller/IndexController.java:15]: this message can be directly in the jsp file

File: src/main/webapp/WEB-INF/views/car/addCar.jsp [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Taxi Service</title>
+</head>
+<body>
+    <h1>To add new car - enter data below</h1>
+    <form method="post" action="${pageContext.request.contextPath}/cars/addCar">
+        Model <input type="text" name="model" required><br>
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/car/addCar.jsp:9]: do you have endpoint with `/cars/addCar` URL?

File: src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp [hunk 9]
```diff
@@ -0,0 +1,14 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Taxi Service</title>
+</head>
+<body>
+    <h1>Add driver to car</h1>
+    <form method="post" action="${pageContext.request.contextPath}/car/addDriverToCar">
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp:8]: same here, check all URLs in the your project

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/java/mate/controller/AddDriverToCarController.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Driver;
+import mate.service.CarService;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/car/driver/add")
```
Comment (Noct2000) [inline src/main/java/mate/controller/AddDriverToCarController.java:15]: image
use the plural, as here, check all URLs

File: src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp [hunk 11]
```diff
@@ -0,0 +1,14 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Taxi Service</title>
+</head>
+<body>
+    <h1>Add driver to car</h1>
+    <form method="post" action="${pageContext.request.contextPath}/addDriverToCar">
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp:8]: don't forget to change URLs in the jsp files after refactoring

File: src/main/webapp/WEB-INF/views/manufacturer/manufacturers.jsp [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+            <td><c:out value="${manufacturer.country}"/></td>
+            <td><a href="${pageContext.request.contextPath}/deleteManufacturer?id=${manufacturer.id}">DELETE</a></td>
+          </tr>
+    </c:forEach>
+    <form action="${pageContext.request.contextPath}/addManufacturer">
+        <input type="submit" value="Add New Manufacturer">
+    </form>
+</table>
+</body>
+</html>
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/manufacturer/manufacturers.jsp:32]: add empty line to the end of the file

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/views/car/addCar.jsp [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Taxi Service</title>
+</head>
+<body>
+    <h1>To add new car - enter data below</h1>
+    <form method="post" action="${pageContext.request.contextPath}/car/add">
+        Model <input type="text" name="model" required><br>
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/car/addCar.jsp:9]:
```suggestion
    <form method="post" action="${pageContext.request.contextPath}/cars/add">
        Model <input type="text" name="model" required><br>
```

**Review by Noct2000 — CHANGES_REQUESTED** [review]

**Review by Noct2000 — APPROVED** [review]

**Pull request closed without merging**